### DISCUSSION TITLE
How to Calculate Colors in WinUI with AccenColor?

### DISCUSSION THREAD

**Discussion by pxinz** [microsoft/microsoft-ui-xaml#9705] [Q&A]
Hi. I am developing a UI library similar to WinUI3 in another programming language, and I want it to adapt to the AccentColor on Windows platforms. I plan to calculate brush colors based on the retrieved AccentColor, but I am encountering some issues: I can't find the specific calculation methods.

I hope to find docs that includes the color calculation formulas, but so far I haven't been successful.

Could anyone provide some guidance or point me towards the relevant documentation?

Thanks!

**Comment by michelgerritse** [comment]
You might want to like at:
[DwmGetColorizationColor](https://learn.microsoft.com/en-us/windows/win32/api/dwmapi/nf-dwmapi-dwmgetcolorizationcolor)

This is, as far as I know, the only documented Win32 API to get the user selected accent color. It's in a strange format (premultiplied alpha ??) so you might need to convert it before using.
